src/bin/cargo/commands/clean.rs

```diff
@@ -6,14 +6,18 @@ use cargo::core::global_cache_tracker::GlobalCacheTracker;
 use cargo::ops::CleanContext;
 use cargo::ops::{self, CleanOptions};
 use cargo::util::print_available_packages;
+use clap_complete::ArgValueCandidates;
 use std::time::Duration;
 
 pub fn cli() -> Command {
     subcommand("clean")
         .about("Remove artifacts that cargo has generated in the past")
         .arg_doc("Whether or not to clean just the documentation directory")
         .arg_silent_suggestion()
-        .arg_clean_package_spec_simple("Package to clean artifacts for")
+        .arg_package_spec_simple(
+            "Package to clean artifacts for",
+            ArgValueCandidates::new(Vec::new), // TODO(completions): implement dynamic completion for `cargo clean -p`
+        )
```

Comment:
Wouldn't this just be `get_dependency_package_candidates`?

Reply:
`cargo clean` accepts any package name in the package tree which is what I thought what `get_dependency_package_candidates` returns